### ISSUE TITLE
Returns discount when I expected it not to

### ISSUE THREAD

**Issue by Luke1982** [tsolucio/PriceCalculation#5]
https://github.com/tsolucio/PriceCalculation/blob/325dea44fc75633fe24392e28c3f694cded4ea92/modules/DiscountLine/DiscountLine.php#L344-L357

At first I thought this was a bug, but it is by design. When I create a discount rule that has *no* category (so set to `--None--`), and relate it to both an Account and Service, **all** services for that account will get that discount.

Studying the code, that is obvious, since it is how it escalates. If it can't find a record where both the service and account are explicitly mentioned, it'll settle for the piece of code I referenced above that will 'settle' for a rule that has a category of '--None--' and an Account, but no product.

This doesn't feel right, since I expect a category of '--None--' to have no effect on the selection procedure, while here it has an explicit effect. Selecting a category will, on the other hand disable the discount behavior. So basically connecting an Account/Contact with a Product/Service won't work through the related lists, since you always need to do it through category.

Basically I think it is too eager to find a result. Where I would think that connecting an Account to a Service is very explicit and would take preference, it actually gives preference to the category. Maybe we could add a category that tells the code to ignore the categories and focus on the related lists.

**Comment by Luke1982** [comment]
OK, I am going to hardfork this module under a different name, in which I will code it like I think it should behave.
